homeassistant/components/smarty/sensor.py

```diff
@@ -0,0 +1,179 @@
+        self._state = self._smarty.supply_air_temperature
+
+
+class ExtractAirTemperatureSensor(SmartySensor):
+    """Extract Air Temperature Sensor."""
+
+    def __init__(self, name, smarty):
+        """Supply Air Temperature Init."""
+        super().__init__(name='{} Extract Air Temperature'.format(name),
+                         device_class=DEVICE_CLASS_TEMPERATURE,
+                         unit_of_measurement=TEMP_CELSIUS,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.extract_air_temperature
+
+
+class OutdoorAirTemperatureSensor(SmartySensor):
+    """Extract Air Temperature Sensor."""
+
+    def __init__(self, name, smarty):
+        """Outdoor Air Temperature Init."""
+        super().__init__(name='{} Outdoor Air Temperature'.format(name),
+                         device_class=DEVICE_CLASS_TEMPERATURE,
+                         unit_of_measurement=TEMP_CELSIUS,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.outdoor_air_temperature
+
+
+class SupplyFanSpeedSensor(SmartySensor):
+    """Supply Fan Speed RPM."""
+
+    def __init__(self, name, smarty):
+        """Supply Fan Speed RPM Init."""
+        super().__init__(name='{} Supply Fan Speed'.format(name),
+                         device_class=None,
+                         unit_of_measurement=None,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.supply_fan_speed
+
+
+class ExtractFanSpeedSensor(SmartySensor):
+    """Extract Fan Speed RPM."""
+
+    def __init__(self, name, smarty):
+        """Extract Fan Speed RPM Init."""
+        super().__init__(name='{} Extract Fan Speed'.format(name),
+                         device_class=None,
+                         unit_of_measurement=None,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.extract_fan_speed
+
+
+class FilterDaysLeftSensor(SmartySensor):
+    """Filter Days Left."""
+
+    def __init__(self, name, smarty):
+        """Filter Days Left Init."""
+        super().__init__(name='{} Filter Days Left'.format(name),
+                         device_class=DEVICE_CLASS_TIMESTAMP,
+                         unit_of_measurement=None,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        days_left = self._smarty.filter_timer
+        self._state = dt_util.now() + dt.timedelta(days=days_left)
```

Comment:
We should make sure that the timestamp doesn't change during the same day, unless `days_left` changes.